src/DurableTask.Netherite/StorageLayer/Faster/FasterKV.cs

```diff
@@ -499,20 +503,39 @@ public override async Task<long> RunCompactionAsync(long target)
                     target - this.Log.BeginAddress,
                     this.GetElapsedCompactionMilliseconds());
 
+                var timeoutTask = Task.Delay(TimeSpan.FromMinutes(10));
```

Comment:
Note: 10 minutes was determined to be a safe timeout window using various kusto queries to determine the expected duration of compaction. But it could be too conservative